core/domain/wipeout_service.py

```diff
@@ -222,6 +220,165 @@ def _generate_activity_to_pseudonymized_ids_mapping(activity_ids):
     }
 
 
+def _delete_improvements_models(user_id):
+    """Delete the improvements models for the user with user_id.
+
+    Args:
+        user_id: str. The id of the user to be deleted.
+    """
+    for model_class in models.Registry.get_storage_model_classes(
+            [models.NAMES.improvements]):
+        if (model_class.get_deletion_policy() not in
+                [base_models.DELETION_POLICY.KEEP,
+                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
+            model_class.apply_deletion_policy(user_id)
+
+
+def _delete_question_models(pending_deletion_request):
```

Comment:
Too much duplication between these functions, IMO. Can you find a way to consolidate?

It's good that you have written comprehensive tests -- that should help you do it safely.